pkg/chunked/cache_linux.go

```diff
@@ -423,14 +436,37 @@ func writeCache(manifest []byte, format graphdriver.DifferOutputFormat, id strin
 		return nil, err
 	}
 
+	fnamesMap := make(map[string]int)
+	getFileNamePosition := func(name string) (int, error) {
+		if pos, found := fnamesMap[name]; found {
+			return pos, nil
+		}
+		pos := fnames.Len()
+		fnamesMap[name] = pos
+
+		nameBytes := []byte(name)
+
+		if err := binary.Write(&fnames, binary.LittleEndian, uint32(len(nameBytes))); err != nil {
+			return 0, err
+		}
+		if _, err := fnames.Write(nameBytes); err != nil {
+			return 0, err
+		}
```

Comment:
You can avoid `[]byte` conversion here, using `name` directly (as in `len(name)` and `fnames.WriteString(name)`)